tahoma_api/client.py

```diff
@@ -121,58 +123,49 @@ async def execute_action_group(
         if mode in supported_modes:
             endpoint = f"{endpoint}/{mode}"
 
-        response = await self.__do_http_request("POST", endpoint, payload)
+        response = await self.__post(endpoint, payload)
 
         return response
 
     async def get_current_execution(self, exec_id: str) -> List[Any]:
         """ Get an action group execution currently running """
-        response = await self.__do_http_request("GET", f"/exec/current/{exec_id}")
+        response = await self.__get(f"/exec/current/{exec_id}")
         # TODO Strongly type executions
 
         return response
 
     async def get_current_executions(self) -> List[Any]:
         """ Get all action groups executions currently running """
-        response = await self.__do_http_request("GET", "/exec/current")
+        response = await self.__get("/exec/current")
         # TODO Strongly type executions
 
         return response
 
-    async def __do_http_request(
-        self, method: str, endpoint: str, payload: Optional[Any] = None
-    ) -> Any:
-        """Make a request to the TaHoma API"""
-        supported_methods = ["GET", "POST"]
-        result = response = None
-
-        if method not in supported_methods:
-            raise Exception
-
-        if method == "GET":
-            async with self.session.get(f"{self.api_url}{endpoint}") as response:
-                result = await response.json()
-
-        if method == "POST":
-            async with self.session.post(
-                f"{self.api_url}{endpoint}", data=payload
-            ) as response:
-                result = await response.json()
-
-        if result is None or response is None:
-            return  # TODO throw error
-
-        # TODO replace by our own library
-        result = humps.decamelize(result)
-
+    async def __get(self, endpoint: str) -> Any:
+        """ Make a GET request to the TaHoma API """
+        async with self.session.get(f"{self.api_url}{endpoint}") as response:
+            await self.check_response(response)
+            return await response.json()
+
+    async def __post(self, endpoint: str, payload: Optional[JSON] = None,) -> Any:
+        """ Make a POST request to the TaHoma API """
+        async with self.session.post(
+            f"{self.api_url}{endpoint}", data=payload
+        ) as response:
+            await self.check_response(response)
+            return await response.json()
```

Comment:
@iMicknl I don't call ` humps.decamelize` for each response. I prefer call it only when needed to avoid useless process.

Reply:
I can understand, since we don't always process the response. For the event listener response you would need it as well (and it is missing now).

Reply:
Sure, we will have to think about it once implemented.